### ISSUE TITLE
web.xml:4: cvc-elt.1: Cannot find the declaration of element 'web-app'

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/google-cloud-eclipse#1166]
Note that this is a false positive. Eclipse is incorrectly flagging an XML file without a DOCTYPE declaration as an error. It shouldn't do this. It's a huge bug in Eclipse, but we may need to work around it anyway. 

```
com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest
testPublishingSubmodules(com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest)  Time elapsed: 8.087 sec  <<< FAILURE!
java.lang.AssertionError: Imported projects have errors
/sox-server/WebContent/WEB-INF/web.xml:4: cvc-elt.1: Cannot find the declaration of element 'web-app'.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:131)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:125)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.importProjects(ProjectUtils.java:112)
	at com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest.setUp(LocalAppEnginePublishOperationTest.java:51)
```

**Comment by elharo** [comment]
Here's another flake in this test at a different place:

https://s3.amazonaws.com/archive.travis-ci.org/jobs/189386937/log.txt

Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.607 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest
testPublishingSubmodules(com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest)  Time elapsed: 2.604 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at org.junit.Assert.assertNotNull(Assert.java:722)
	at com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEnginePublishOperationTest.setUp(LocalAppEnginePublishOperationTest.java:65)

**Comment by chanseokoh** [comment]
The second error is completely different from the first error in nature. However, I can tell I have encountered the second error many times. I've wondered how this could even happen, and I actually attempted to find a reason a few times in the past. Now I see one culprit: `ProjectUtils.importProjects()` should call `waitForProjects(projects)` instead of `waitForProjects()`.

**Comment by briandealwis** [comment]
> Now I see one culprit: ProjectUtils.importProjects() should call waitForProjects(projects) instead of waitForProjects().

:blush: I thought I had logic in there to treat no-projects as the entire workspace ⚡️

**Comment by chanseokoh** [comment]
For the second error, this is the right issue: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1798

**Comment by chanseokoh** [comment]
Another cvc-elt.1 XML failure in a difference place:

```
testHelloWorld_nonDefaultLocation(com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest)  Time elapsed: 20.031 sec  <<< FAILURE!
java.lang.AssertionError: New Maven project has errors
/appWithPackageProjectInTemp/pom.xml:4: cvc-elt.1: Cannot find the declaration of element 'project'.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:147)
	at com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest.createAndCheck(NewMavenBasedAppEngineProjectWizardTest.java:101)
	at com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest.testHelloWorld_nonDefaultLocation(NewMavenBasedAppEngineProjectWizardTest.java:68)
```

**Comment by briandealwis** [comment]
This should have been fixed by #2086 and #2092.

**Comment by briandealwis** [comment]
Just saw this failure on Oxygen:
```
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 49.586 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 34.698 sec  <<< FAILURE!
java.lang.AssertionError: Imported App Engine standard Java 7 project has errors
/AESv7/src/main/webapp/WEB-INF/web.xml:5: cvc-elt.1: Cannot find the declaration of element 'web-app'.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:167)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:69)
```
